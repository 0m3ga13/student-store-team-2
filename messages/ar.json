{
    "Index": {
        "Title": "المتجر الصغير",
        "Home": "الرئيسية",
        "Products": "المنتجات",
        "Orders": "الطلبيات",
        "State": "الحالة",
        "Buy": "شراء",
        "Borrow": "إعارة",
        "Description": "هذه المنصة هي مربحة للجميع: القرضاء يحققون دخلاً إضافيًا، والمقترضون يقللون التكاليف.",
        "Profile": "الحسـاب",
        "For Sale": "للبيع",
        "For Borrow": "للإعارة",
        "Sale & Borrow": "إعارة وشراء",
        "Day": "لليوم",
        "How It Works": "طريـقة العمل",
        "Hello": "مرحبا",
        "Sign out": "تسجيل الخروج",
        "Sign In": "تسجيل الدخول",
        "Login with Google": "سحل الدخول بحسابك",
        "Sign Up With Google": "سحل الدخول بحسابك"
    }
}
