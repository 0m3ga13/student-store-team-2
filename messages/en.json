{
    "Index": {
        "Title": "LaCité",
        "Home": "Home",
        "Products": "Products",
        "Orders": "Orders",
        "State": "State",
        "Donate": "Donate",
        "Buy": "Buy",
        "Borrow": "Borrow",
        "Description": "This platform is a win-win for everyone involved lenders earn extra income, borrowers cut costs.",
        "Profile": "Profile",
        "for_sale": "for_sale",
        "for_borrow": "for_borrow",
        "Sale & Borrow": "Sale & Borrow",
        "Day": "Day",
        "How It Works": "How It Works",
        "recently added": "recently added",
        "Hello": "Hello",
        "Sign out": "Sign out",
        "Sign In": "Sign In",
        "Login with Google": "Login with Google",
        "Sign Up With Google": "Sign Up With Google",
        "Comprehensive Catalog": "Comprehensive Catalog",
        "feature descriprion 1": "Access a diverse range of academic essentials for various student needs.",
        "Buy, Sell, Borrow": "Buy, Sell, Borrow",
        "feature descriprion 2": "Seamlessly buy, sell, or borrow items for flexible access to materials.",
        "User-Friendly Experience": "User-Friendly Experience",
        "feature descriprion 3": "Enjoy easy navigation and quick access to academic resources.",
        "Community Support Initiative": "Community Support Initiative",
        "feature descriprion 4": "Engage in supporting fellow students through dedicated programs, fostering a sense of community.",
        "My Dashboard": "My Dashboard",
        "Awaiting Pickup": "Awaiting Pickup",
        "All": "All",
        "Active": "Active",
        "Completed": "Completed",
        "Sort": "Sort:",
        "Newest": "Newest",
        "Oldest": "Oldest",
        "My Stuff": "My Stuff",
        "borrowings": "borrowings",
        "lendings": "lendings",
        "purchases": "purchases",
        "sales": "sales",
        "My Borrowings": "My Borrowings",
        "My Lendings": "My Lendings",
        "My Purchases": "My Purchases",
        "My Sales": "My Sales",
        "General details": "General details",
        "First Name": "First Name",
        "Last Name": "Last Name",
        "Birth Date": "Birth Date",
        "Gender": "Gender",
        "Institution": "Institution",
        "Contact & address Detail": "Contact & address Detail",
        "Phone Number": "Phone Number",
        "Building": "Building",
        "Street": "Street",
        "City": "City",
        "State_place": "State",
        "Save Changes": "Save Changes",
        "Upload Image": "Upload Image",
        "Close": "Close",
        "Upload": "Upload",
        "Loading": "Loading",
        "Not a member ?": "Not a member ?",
        "Sign Up": "Sign Up",
        "Store": "Store",
        "Create Your": "Create Your",
        "and start exchanging with other students": "and start exchanging with other students",
        "Or Login Using:": "Or Login Using:",
        "Login": "Login",
        "LOGIN": "LOGIN ",
        "Already a member ?": "Already a member ?",
        "Email and password are required": "Email and password are required",
        "Password should be at least 8 characters long": "Password should be at least 8 characters long",
        "and Conditions and Privacy Policy of The website": "and Conditions and Privacy Policy of The website",
        "Email": "Email",
        "Password": "Password",
        "Full Name": "Full Name",
        "Address": "Address",
        "and Privacy Policy": "and Privacy Policy",
        "I have read and agree with all of the Terms": "I have read and agree with all of the Terms",
        "PROFILE": "PROFILE",
        "SECURITY": "SECURITY",
        "Upload ID": "Upload ID",
        "Edit ID": "Edit ID",
        "Edit Image": "Edit Image",
        "WHY DONATE": "WHY DONATE",
        "MAKE A DIFFERENCE BY DONATING": "MAKE A DIFFERENCE BY DONATING",
        "Donate paragraph": "In our fast-paced lives, it's easy to forget the power of gratitude. Students, often on tight budgets, rely on our support. Your donation is more than just financial aid; it's a symbol of gratitude. By giving generously, you're helping students create a brighter future, one contribution at a time.",
        "Small Help": "Small Help",
        "Some Help": "Some Help",
        "Big Help": "Big Help",
        "List an Item": "List an Item",
        "Drag and drop some files here or click to select files": "Drag and drop some files here or click to select files",
        "Location": "Location",
        "Category": "Category",
        "books": "books",
        "kitchen": "kitchen",
        "sports": "sports",
        "games": "games",
        "clothes": "clothes",
        "furniture": "furniture",
        "description": "Description",
        "Product Condition": "Product Condition",
        "Sell": "Sell",
        "Price Per Week": "Price Per Week",
        "Price": "Price",
        "Upload Image(s)": "Upload Image(s)",
        "List": "List",
        "Product Name": "Product Name",
        "Comments": "Comments",
        "Type your comment": "Type your comment",
        "Type your reply": "Type your reply",
        "Respond": "Respond",
        "Like": "Like",
        "Send": "Send",
        "Authentication error": "Invalid login credentials. Please check your email and password.",
        "Filter": "Filter",
        "Pending": "Pending",
        "Quick Links": "Quick Links",
        "About LACITÉ": "ABOUT LACITÉ",
        "About Us": "ABOUT US",
        "Github Project Link": "GITHUB PROJECT LINK",
        "Contact Us": "Contact Us",
        "We Couldn't Find the page you're looking for": "We Couldn't Find the page you're looking for",
        "Go To Home": "Go To Home",
        "We Couldn't Find the requested Product": "We Couldn't Find the requested Product.",
        "Welcome to Our Student Store": "Welcome to Our Student Store",
        "Your one-stop destination for all your academic needs!": "Your one-stop destination for all your academic needs!",
        "Whether you're searching for textbooks, calculators, lab equipment, or even selling items you no longer need, we've got you covered Our store offers a variety of options to buy, sell, and even borrow items, ensuring you have access to the materials you need for your academic journey": "Whether you're searching for textbooks, calculators, lab equipment, or even selling items you no longer need, we've got you covered. Our store offers a variety of options to buy, sell, and even borrow items, ensuring you have access to the materials you need for your academic journey",
        "Why Choose LaCité?": "Why Choose LaCité?",
        "Our student store stands out for its diverse range of resources and services Whether you're looking to buy or sell textbooks, scientific equipment, or other academic necessities, we've created a platform that caters specifically to students' needs; Our commitment to providing a seamless experience ensures that you can easily find what you're looking for and even explore options to borrow items you may need temporarily; We prioritize your academic success by offering convenience and accessibility at every step": "Our student store stands out for its diverse range of resources and services. Whether you're looking to buy or sell textbooks, scientific equipment, or other academic necessities, we've created a platform that caters specifically to students' needs. Our commitment to providing a seamless experience ensures that you can easily find what you're looking for and even explore options to borrow items you may need temporarily. We prioritize your academic success by offering convenience and accessibility at every step.",
        "Textbooks": "Textbooks",
        "Stationery": "Stationery",
        "Art Supplies": "Art Supplies",
        "Electronics": "Electronics",
        "Study Guides": "Study Guides",
        "And More": "And More",
        "Start exploring today!": "Start exploring today!",
        "Find all the tools and resources you need for your academic journey From textbooks to stationery, we've got you covered": "Find all the tools and resources you need for your academic journey. From textbooks to stationery, we've got you covered.",
        "Get Started": "Get Started",
        "Meet the": "Meet the",
        "team": "team",
        "Support Students": "Support Students",
        "About": "About",
        "Technologies Used": "Technologies Used",
        "Books": "Books",
        "Explore a vast collection of books covering various genres, from thrilling mysteries to captivating stories that transport you to different worlds Dive into these literary adventures and let your imagination soar": "Explore a vast collection of books covering various genres, from thrilling mysteries to captivating stories that transport you to different worlds. Dive into these literary adventures and let your imagination soar.",
        "Sports Equipments": "Sports Equipments",
        "Find top-quality sports equipment to elevate your game, whether you're passionate about team sports or prefer individual athletic pursuits Discover gear designed to enhance your performance and support your active lifestyle": "Find top-quality sports equipment to elevate your game, whether you're passionate about team sports or prefer individual athletic pursuits. Discover gear designed to enhance your performance and support your active lifestyle.",
        "Clothes": "Clothes",
        "Express your style with a diverse range of clothing options, from casual wear to formal attire Discover fashion that suits your personality, trends that inspire, and comfortable, stylish outfits for any occasion": "Express your style with a diverse range of clothing options, from casual wear to formal attire. Discover fashion that suits your personality, trends that inspire, and comfortable, stylish outfits for any occasion.",
        "Games": "Games",
        "Discover a world of entertainment with an array of games that cater to various interests and ages From stimulating puzzles to thrilling adventures, these games promise endless fun and excitement for everyone": "Discover a world of entertainment with an array of games that cater to various interests and ages. From stimulating puzzles to thrilling adventures, these games promise endless fun and excitement for everyone.",
        "Explore cutting-edge electronics that simplify your life and elevate your technological experiences From innovative gadgets to essential devices, find the latest tech solutions here": "Explore cutting-edge electronics that simplify your life and elevate your technological experiences. From innovative gadgets to essential devices, find the latest tech solutions here.",
        "Kitchen Cookwares": "Kitchen Cookwares",
        "Enhance your culinary adventures with high-quality kitchen cookware From utensils to appliances, discover tools that make cooking enjoyable and efficient, allowing you to create delicious meals with ease": "Enhance your culinary adventures with high-quality kitchen cookware. From utensils to appliances, discover tools that make cooking enjoyable and efficient, allowing you to create delicious meals with ease.",
        "See More": "See More",
        "Browse & Choose": "Browse & Choose",
        "Connect with Sellers": "Connect with Sellers",
        "Confirm Purchase": "Confirm Purchase",
        "Register and unlock access to our diverse collection of items available for borrowing or purchase It only takes a few moments to create your account": "Register and unlock access to our diverse collection of items available for borrowing or purchase. It only takes a few moments to create your account.",
        "Explore our extensive range of products Whether it's books, electronics, furniture, gaming gear, or everyday essentials, take your time to browse through our offerings Once you find what you need, add it to your cart": "Explore our extensive range of products. Whether it's books, electronics, furniture, gaming gear, or everyday essentials, take your time to browse through our offerings. Once you find what you need, add it to your cart.",
        "Contact the sellers directly to discuss details, ask questions, and negotiate the deal This step ensures you get all the information you need before finalizing your purchase or borrowing arrangement": "Contact the sellers directly to discuss details, ask questions, and negotiate the deal. This step ensures you get all the information you need before finalizing your purchase or borrowing arrangement.",
        "Once you've settled all the details and agreed with the seller, confirm your purchase or borrowing Make the payment or arrange the pickup/delivery, and you're all set to enjoy your new item!": "Once you've settled all the details and agreed with the seller, confirm your purchase or borrowing. Make the payment or arrange the pickup/delivery, and you're all set to enjoy your new item!",
        "Enjoy!": "Enjoy!",
        "How it Works": "How it Works",
        "They Said About Us": "They Said About Us",
        "for sale": "Buy",
        "for borrow": "Borrow",
        "Filter By": "Filter By",
        "For Borrow":"To Borrow",
        "For Sale": "To Buy",
        "Price Filter": "Price Filter",
        "Product Status": "Product Status",
        "Categories": "Categories",
        "Product Note": "Product Condition",
        "Reset": "Reset",
        "Required": "Required",
        "Male": "Male",
        "Female": "Female",
        "Enter Location": "Enter Location",
        "Category:": "Category:",
        "Contact Details": "Contact Details",
        "Additionnal Information": "Additionnal Information",
        "Notifications": "Notifications",
        "Add Product": "Add Product",
        "Search": "Search",
        "Logout": "Logout",
        "Settings": "Settings",
        "Dashboard": "Dashboard",
        "bought form": "Bought from",
        "sold to": "Sold to",
        "borrowed from": "Borrowed from",
        "borrowed to": "Borrowed to",
        "Name": "Name",
        "Message": "Message",
        "Borrow Period": "Borrow Period",
        "Status": "Status",
        "Actions": "Actions",
        "my stuff": "my stuff",
        "requested": "has requested to buy your item",
        "accept": "has accepeted your request",
        "reject": "has rejected your request",
        "abort": "has aborted the order",
        "complete": "has picked your item",
        "View": "View",
        "Requested": "Requested",
        "Search for a product": "Search for a product",
        "Forgot your password?": "Forgot your password?",
        "Remember Me": "Remember Me"
        
    }
}
