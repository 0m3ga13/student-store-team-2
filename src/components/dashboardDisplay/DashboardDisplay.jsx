"use client";

import Link from "next/link";
import { useTranslations } from "next-intl";
import { useEffect, useState } from "react";
import { GoPlus } from "react-icons/go";

import DashboardTable from "../tables/dashboardTable/DashboardTable";
import UserStuffTable from "../tables/userStuffTable/UserStuffTable";

const DashboardDisplay = ({ dashboardData, type }) => {
    const t = useTranslations("Index");
    const [data, setData] = useState(dashboardData);

    const handleChange = (e) => filterBySearch(e.target.value);

    const filterBySearch = (search) => {
        const array = dashboardData.filter((item) => {
            if (!search) return item;
            else {
                return (
                    item.name.toLowerCase().includes(search.toLowerCase()) ||
                    item.category_name
                        .toLowerCase()
                        .includes(search.toLowerCase()) ||
                    item.raw_user_meta_data.full_name
                        .toLowerCase()
                        .includes(search.toLowerCase())
                );
            }
        });
        setData(array);
    };

    const filterByStatus = (e) => {
        const status = e.target.value.toLowerCase();
        if (status === "all") setData(dashboardData);
        else {
            const array = dashboardData.filter(
                (item) => item.status === status
            );

            setData(array);
        }
    };

    useEffect(() => {
        setData(dashboardData);
    }, [dashboardData, type]);

    return (
        <div className='min-h-screen'>
            <h1 className='text-3xl capitalize font-lato font-medium ml-4 mb-6'>
                {type || "my stuff"}
            </h1>
            <div className='mb-8'>
                <div className='flex items-end'>
                    <select
                        name=''
                        id=''
                        onChange={filterByStatus}
                        className='bg-white py-2 px-4 w-16 h-12 border border-gray-300 rounded-md  ml-4'
                    >
                        <option value='all' defaultChecked>
                            {t("All")}
                        </option>
                        <option value='requested' defaultChecked>
                            {t("Requested")}
                        </option>
                        <option value='pending' defaultChecked>
                            {t("Pending")}
                        </option>
                        <option value='completed' defaultChecked>
                            {t("Completed")}
                        </option>
                    </select>
                    <form className='ml-4  inline-block'>
                        <label className='sr-only'>{t("Search")}</label>
                        <div className='mt-1 relative lg:w-64 xl:w-96'>
                            <input
                                type='text'
                                onChange={handleChange}
                                className='bg-gray-50 h-12 border border-gray-300 text-gray-900 sm:text-sm rounded-md focus:ring-accent focus:border-accent block w-full p-2.5'
                                placeholder={t("Search for a product")}
                            />
                        </div>
                    </form>
                    <Link
                        href='dashboard?type=add product'
                        className='ml-8 sm:ml-auto rounded-md p-2.5  bg-cyan-600 text-white flex items-center hover:bg-cyan-800'
                    >
                        <GoPlus className='w-7 h-7' />
                        <span>{t("Add Product")}</span>
                    </Link>
                </div>
            </div>
            {type === "stuff" ? (
                <UserStuffTable data={data} />
            ) : (
                <DashboardTable type={type} data={data} />
            )}
        </div>
    );
};

export default DashboardDisplay;
