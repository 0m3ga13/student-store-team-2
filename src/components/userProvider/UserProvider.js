"use client";

import { createContext, useContext, useEffect, useState } from "react";

const UserContext = createContext();

export function UserProvider({ children, getCurrentUser }) {
    const [user, setUser] = useState(null);
    const [userData, setUserData] = useState(null);
    const [loading, setLoading] = useState(true);
    useEffect(() => {
        const fetchData = async () => {
            try {
                const data = await getCurrentUser();
                setUser(data?.user);
            } catch (error) {
                console.error(error);
            }
        };

        fetchData();
    }, []);

    useEffect(() => {
        const fetchUserData = async () => {
            try {
                if (user) {
                    await setUserData(user.user_metadata);
                    setLoading(false);
                } else {
                    await new Promise((resolve) => setTimeout(resolve, 1000));

                    setLoading(false);
                }
            } catch (error) {
                // Handle errors
                setLoading(false);
                throw error;
            }
        };

        fetchUserData();
    }, [user]);

    return (
        <UserContext.Provider value={{ user, userData, loading }}>
            {children}
        </UserContext.Provider>
    );
}

export function useUser() {
    return useContext(UserContext);
}
