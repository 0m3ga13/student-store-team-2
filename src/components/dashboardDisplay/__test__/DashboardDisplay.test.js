import { NextIntlClientProvider } from "next-intl";
import renderer from "react-test-renderer";

import { UserProvider } from "@/components/userProvider/UserProvider";

import DashboardDisplay from "../DashboardDisplay";
import messages from "../../../../messages/en.json";
jest.mock("firebase/auth", () => {
    const authInstance = {
        // Add mock methods and properties as needed
    };

    return {
        getAuth: jest.fn(() => authInstance),
        createUserWithEmailAndPassword: jest.fn(),
        signInWithEmailAndPassword: jest.fn(),
        // Add other authentication methods used in your code
        GoogleAuthProvider: jest.fn(),
    };
});
jest.mock("firebase/firestore", () => {
    return {
        getFirestore: jest.fn(),
        doc: jest.fn(),
        setDoc: jest.fn(),
        // Add other Firestore methods used in your code
    };
});

it("renders correctly", () => {
    const tree = renderer
        .create(
            <NextIntlClientProvider locale='en' messages={messages}>
                <UserProvider>
                    <DashboardDisplay />
                </UserProvider>
            </NextIntlClientProvider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
