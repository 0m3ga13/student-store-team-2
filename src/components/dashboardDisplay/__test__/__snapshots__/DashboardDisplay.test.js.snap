// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="min-h-screen"
>
  <h1
    className="text-3xl capitalize font-lato font-medium ml-4 mb-6"
  >
    my stuff
  </h1>
  <div
    className="mb-8"
  >
    <div
      className="flex items-end"
    >
      <select
        className="bg-white py-2 px-4 w-16 h-12 border border-gray-300 rounded-md  ml-4"
        id=""
        name=""
        onChange={[Function]}
      >
        <option
          defaultChecked={true}
          value="all"
        >
          All
        </option>
        <option
          defaultChecked={true}
          value="requested"
        >
          Requested
        </option>
        <option
          defaultChecked={true}
          value="pending"
        >
          Pending
        </option>
        <option
          defaultChecked={true}
          value="completed"
        >
          Completed
        </option>
      </select>
      <form
        className="ml-4  inline-block"
      >
        <label
          className="sr-only"
        >
          Search
        </label>
        <div
          className="mt-1 relative lg:w-64 xl:w-96"
        >
          <input
            className="bg-gray-50 h-12 border border-gray-300 text-gray-900 sm:text-sm rounded-md focus:ring-accent focus:border-accent block w-full p-2.5"
            onChange={[Function]}
            placeholder="Search for a product"
            type="text"
          />
        </div>
      </form>
      <a
        className="ml-8 sm:ml-auto rounded-md p-2.5  bg-cyan-600 text-white flex items-center hover:bg-cyan-800"
        href="dashboard?type=add product"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <svg
          className="w-7 h-7"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11.75 4.5a.75.75 0 0 1 .75.75V11h5.75a.75.75 0 0 1 0 1.5H12.5v5.75a.75.75 0 0 1-1.5 0V12.5H5.25a.75.75 0 0 1 0-1.5H11V5.25a.75.75 0 0 1 .75-.75Z"
          />
        </svg>
        <span>
          Add Product
        </span>
      </a>
    </div>
  </div>
  <table
    className="table-fixed min-w-full divide-y divide-gray-200 "
  >
    <thead
      className="bg-gray-100"
    >
      <tr>
        <th
          className="p-4 text-left text-xs font-medium text-gray-500 uppercase"
          scope="col"
        >
          Name
        </th>
        <th
          className="p-4 text-left text-xs font-medium text-gray-500 uppercase"
          scope="col"
        />
        <th
          className="p-4 text-left text-xs font-medium text-gray-500 uppercase"
          scope="col"
        >
          Message
        </th>
        <th
          className="p-4 text-left text-xs font-medium text-gray-500 uppercase"
          scope="col"
        >
          Phone Number
        </th>
        <th
          className="p-4 text-left text-xs font-medium text-gray-500 uppercase"
          scope="col"
        >
          Status
        </th>
        <th
          className="p-4 text-left text-xs font-medium text-gray-500 uppercase"
          scope="col"
        >
          Actions
        </th>
      </tr>
    </thead>
    <tbody
      className="bg-white divide-y divide-gray-200 "
    />
  </table>
</div>
`;
